dynamiq/memory/backend/pinecone.py

```diff
@@ -1,153 +1,150 @@
-            )
-            messages = [
-                Message(
-                    content=match.metadata["content"],
-                    role=match.metadata["role"],
-                    metadata={k: v for k, v in match.metadata.items() if k not in ("content", "role")},
-                )
-                for match in query_response["matches"]
-            ]
+            documents = self.vector_store.list_documents(include_embeddings=False)
+            messages = [self._document_to_message(doc) for doc in documents]
             return sorted(messages, key=lambda msg: msg.metadata.get("timestamp", 0))
         except Exception as e:
             raise PineconeError(f"Error retrieving messages from Pinecone: {e}") from e
 
+    def _prepare_filters(self, filters: dict = None) -> dict:
+        """Convert simple filters to Pinecone filter format."""
+        if not filters:
+            return None
+
+        if all(isinstance(v, (str, int, float, bool)) for v in filters.values()):
+            conditions = []
+            for key, value in filters.items():
+                conditions.append({"field": key, "operator": "==", "value": value})
+            return {"operator": "AND", "conditions": conditions}
+        return filters
+
     def search(self, query: str = None, filters: dict = None, limit: int = None) -> list[Message]:
         """Searches for messages in Pinecone based on the query and/or filters."""
-        limit = limit or self.config.search_limit
-        normalized_filters = self._normalize_filters(filters) if filters else None
         try:
+            normalized_filters = self._prepare_filters(filters)
+
             if query:
                 embedding_result = self.embedder.embed_text(query)
-                embedding = embedding_result["embedding"]
-                response = self._index.query(
-                    vector=embedding,
-                    top_k=limit,
-                    include_metadata=True,
-                    filter=normalized_filters,
+                documents = self.vector_store._embedding_retrieval(
+                    query_embedding=embedding_result["embedding"],
+                    namespace=self.namespace,  # Add namespace
+                    filters=normalized_filters,
+                    top_k=limit or 10,
+                    exclude_document_embeddings=True,
                 )
             elif normalized_filters:
                 dummy_vector = [0.0] * self.embedder.dimensions
-                response = self._index.query(
-                    vector=dummy_vector,
-                    top_k=limit,
-                    include_metadata=True,
-                    filter=normalized_filters,
+                documents = self.vector_store._embedding_retrieval(
+                    query_embedding=dummy_vector,
+                    namespace=self.namespace,  # Add namespace
+                    filters=normalized_filters,
+                    top_k=limit or 10,
```

Comment:
Let's move this constant to the function's default arguments.
```suggestion
                    top_k=limit,
```

Reply:
Please update it in the next PR!)